projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-detail-actions/order-detail-actions.component.ts

```diff
@@ -9,5 +10,7 @@ import { OrderDetailsService } from '../order-details.service';
 export class OrderDetailActionsComponent {
   constructor(protected orderDetailsService: OrderDetailsService) {}
 
-  order$: Observable<any> = this.orderDetailsService.getOrderDetails();
+  order$: Observable<any> = this.orderDetailsService
+    .getOrderDetails()
+    .pipe(filter((order) => Object.keys(order).length > 0));
```

Comment:
To make it bulletproof we may use `map` here (map every emission to null on undefined if order object is empty, instead of not emitting anything)